x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/rule_add.test.tsx

```diff
@@ -195,51 +194,48 @@ describe.skip('rule_add', () => {
     actionTypeRegistry.list.mockReturnValue([actionTypeModel]);
     actionTypeRegistry.has.mockReturnValue(true);
 
-    wrapper = mountWithIntl(
-      <QueryClientProvider client={new QueryClient()}>
-        <RuleAdd
-          consumer={ALERTING_FEATURE_ID}
-          onClose={onClose}
-          initialValues={initialValues}
-          onSave={() => {
-            return new Promise<void>(() => {});
-          }}
-          actionTypeRegistry={actionTypeRegistry}
-          ruleTypeRegistry={ruleTypeRegistry}
-          metadata={{ test: 'some value', fields: ['test'] }}
-          ruleTypeId={ruleTypeId}
-          validConsumers={validConsumers}
-        />
-      </QueryClientProvider>
-    );
-
-    // Wait for active space to resolve before requesting the component to update
-    await act(async () => {
-      await nextTick();
-      wrapper.update();
-    });
+    return {
+      consumer: ALERTING_FEATURE_ID,
+      onClose,
+      initialValues,
+      onSave: () => {
+        return new Promise<void>(() => {});
+      },
+      actionTypeRegistry,
+      ruleTypeRegistry,
+      metadata: { test: 'some value', fields: ['test'] },
+      ruleTypeId,
+      validConsumers,
+    };
   }
 
   it('renders rule add flyout', async () => {
     (fetchUiConfig as jest.Mock).mockResolvedValue({
       minimumScheduleInterval: { value: '1m', enforce: false },
     });
+
     const onClose = jest.fn();
-    await setup({
+    const props = await setup({
       initialValues: {},
       onClose,
     });
 
-    await act(async () => {
-      await nextTick();
-      wrapper.update();
+    render(
+      <IntlProvider locale="en">
+        <QueryClientProvider client={new QueryClient()}>
+          <RuleAdd {...props} />
+        </QueryClientProvider>
+      </IntlProvider>
+    );
+
+    await waitFor(() => {
+      return expect(screen.getByTestId('addRuleFlyoutTitle')).toBeInTheDocument();
     });
 
-    expect(wrapper.find('[data-test-subj="addRuleFlyoutTitle"]').exists()).toBeTruthy();
-    expect(wrapper.find('[data-test-subj="saveRuleButton"]').exists()).toBeTruthy();
-    expect(wrapper.find('[data-test-subj="showRequestButton"]').exists()).toBeTruthy();
+    expect(screen.getByTestId('saveRuleButton')).toBeInTheDocument();
```

Comment:
I would also change all these occurrences of `screen.getByTestId` to `await screen.findByTestId`